src/coreclr/jit/emitarm64.cpp

```diff
@@ -2016,21 +2021,31 @@ void emitter::emitInsSanityCheck(instrDesc* id)
             break;
 
         case IF_SVE_EB_1A: // ........xx...... ..hiiiiiiiiddddd -- SVE broadcast integer immediate (unpredicated)
+            imm = emitGetInsSC(id);
             assert(insOptsScalableStandard(id->idInsOpt()));
-            // Size specifier must be able to fit left-shifted immediate
-            assert(insOptsScalableAtLeastHalf(id->idInsOpt()) || !id->idOptionalShift());
             assert(isVectorRegister(id->idReg1()));                           // ddddd
-            assert(isValidSimm8(emitGetInsSC(id)));                           // iiiiiiii
             assert(isValidVectorElemsize(optGetSveElemsize(id->idInsOpt()))); // xx
+
+            if (!isValidSimm8(imm))
+            {
+                // Size specifier must be able to fit a left-shifted immediate
+                assert(isValidSimm8_MultipleOf256(imm)); // iiiiiiii
+                assert(insOptsScalableAtLeastHalf(id->idInsOpt()));
+            }
```

Comment:
can you add an assert in the `else` clause about `id->idInsOpt() == NONE` or whatever relevant?

Reply:
If we encode `imm` in `emitIns`, this logic ends up going away.